### ISSUE TITLE
Add a dynamic integration of srcML into the add_srcML function

### ISSUE THREAD

**Issue by nalmadi** [nalmadi/EMIP-Toolkit#50]
The add_srcML currently uses pre-generated files for the EMIP dataset code.  It does not generate srcML tags for any piece of code. 

It would be great to integrate srcML into the tool so it is called automatically (behind the scene) to generate the srcML tags for any code, then add the tags to the dataframe.

This means that we will add srcML as a dependency, so let's see if we can do this in an easy way.  Not sure if srcML is downloadable through pip or similar.  This might create problems for our automated Action testing, if srcML is not downloadable through pip.

A good start would be at srcML website to understand the tool and how it works: https://www.srcml.org/

**Comment by quanphan2906** [comment]
I don't think it can be downloaded via pip, because there is no search result for `srcml` on PyPi: https://pypi.org/search/?q=srcml. There is a package called `srcml` on one of Anaconda's channels, but there was no documentation or even any introduction to what the package tries to do. You can take a look: https://anaconda.org/yijunyu/srcml.

How should we proceed in this situation?

**Comment by nalmadi** [comment]
Good questions, I think pip is mostly for Python packages so I doubt it will be there.

Maybe you should ask here: https://github.com/srcML/srcML
